src/org/thoughtcrime/securesms/qr/QrActivity.java

```diff
@@ -108,6 +140,24 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case android.R.id.home:
                 finish();
                 return true;
+            case R.id.share:
+                qrShowFragment.shareQr();
+                break;
+            case R.id.copy:
+                qrShowFragment.copyQrData();
+                break;
+            case R.id.load_from_image:
+                Permissions.with(this)
+                           .request(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                           .ifNecessary()
+                           .withPermanentDenialDialog(getString(R.string.perm_explain_access_to_storage_denied))
+                           .onAllGranted(() -> Util.selectMediaType(this, "image/*", null, REQUEST_CODE_IMAGE))
+                           .execute();
```

Comment:
We can freely decide what we want to do here; we don't need the permission to select an image, but we might need the permission later on when we want to show the recent photo rail.

What you will need to do is replace `WRITE_EXTERNAL_STORAGE` with `READ_EXTERNAL_STORAGE`, but why not use `selectImage()` instead of this whole block? It's always better not to have code duplication.

Reply:
now that you say it I don't even know why I moved `selectMediaType` to utils instead of using AttachmentManager directly :sweat: :man_facepalming:  will fix that

Reply:
>  but why not use `selectImage()` instead of this whole block?

done, thanks for pointing that out!